File: apps/ios/Sources/SchedulerApp/Models/Schedule.swift
import Foundation

struct Schedule: Identifiable, Codable, Hashable {
    let id: String
    let tenantId: String
    let name: String
    let startDate: Date
    let endDate: Date
    let shifts: [Shift]
    let status: ScheduleStatus
    let createdAt: Date
    let updatedAt: Date
    // Defaulted so existing constructions keep compiling; "evenings" maps to the API's "night".
    var settings: ScheduleSettings = ScheduleSettings()
}

struct ScheduleSettings: Codable, Hashable {
    var mornings: Bool = false
    var afternoons: Bool = false
    var evenings: Bool = false
    var timezone: String = "UTC"
}

struct Shift: Identifiable, Codable, Hashable {
    let id: String
    let scheduleId: String
    var dayOfWeek: DayOfWeek
    var startTime: String
    var endTime: String
    let assignedWorkerId: String?
    let stationId: String?
    let notes: String?
}

enum DayOfWeek: String, Codable, CaseIterable {
    case monday, tuesday, wednesday, thursday, friday, saturday, sunday
}

enum ScheduleStatus: String, Codable {
    case draft
    case published
    case archived
}

File: apps/android/app/src/main/java/com/schedulersystems/scheduler/models/domain/Schedule.kt
package com.schedulersystems.scheduler.models.domain

import java.time.Instant
import java.time.LocalTime

data class Schedule(
    val id: String,
    val name: String,
    val tenantId: String,
    val employees: List<Employee>,
    val currentPriorities: List<String>,
    val settings: ScheduleSettings,
    val nextSchedule: List<ShiftRow>,
    val createdAt: Instant,
    val updatedAt: Instant,
    // "active" | "archived" — mirrors the Go API schedule status. Defaulted so existing
    // constructors keep working; used to split My Schedules (active) from Archived.
    val status: String = "active"
)

data class Employee(
    val id: String,
    val name: String,
    val email: String?,
    val phone: String?,
    val role: Role,
    val priorityMap: Map<String, Int>
)

data class Shift(
    val day: String,
    val startTime: LocalTime,
    val endTime: LocalTime,
    val assignedWorker: String?
)

data class ShiftRow(
    val shifts: List<Shift>
)

data class ScheduleSettings(
    val submissionDeadline: SubmissionDeadline?,
    val enabledShifts: EnabledShifts,
    val timezone: String
)

data class SubmissionDeadline(
    val enabled: Boolean,
    val deadline: Instant?
)

data class EnabledShifts(
    val mornings: Boolean,
    val afternoons: Boolean,
    val evenings: Boolean
)

enum class Role {
    EMPLOYER,
    EMPLOYEE,
    ADMIN
}
